src/Compilers/Core/Portable/Operations/OperationExtensions.Clone.cs

```diff
@@ -0,0 +1,477 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    public static partial class OperationExtensions
+    {
+        private class Cloner : OperationVisitor<object, IOperation>
```

Comment:
code doing cloning is safer and easier than hand writing new LocalReference .. and giving right values in. if the argument is whether we should write cloner or not, then you are right, just typing these 3 are less work. but cloner is already written. so I am not sure why it is not easier.